lingua_franca/lang/parse_en.py

```diff
@@ -342,7 +346,8 @@ def _extract_whole_number_with_text_en(tokens, short_scale, ordinals):
         next_word = tokens[idx + 1].word if idx + 1 < len(tokens) else ""
 
         if is_numeric(word[:-2]) and \
-                (word.endswith("st") or word.endswith("nd") or word.endswith("rd") or word.endswith("th")):
+                (word.endswith("st") or word.endswith("nd") or
```

Comment:
As long as this bit's turned green, I'd like to evangelize for `any()` and `all()` wherever they're logical, because much of our code lives in loops, over input of arbitrary length.

```python
has_ordinal_suffix = (word.endswith(suffix) for suffix in ["st", "nd", "rd", "th"])
any(has_ordinal_suffix)
```

just as an "as long as you're in there" thing. It looks tiny every time we do it, but it'll add up. Doesn't work everywhere, but it works in many places.

Reply:
i think that in general those changes should be part of a different PR (code cleanup PR?), these green bits are just pep8 fixes pycharm does automatically.... if we are touching actual logic i'd rather have it reviewed on it's own rather than lost in an unrelated feature. We want to make it easy to track and revert changes that cause bugs, Unmerge PR is easier than looking for specific commits. so while this is a tiny thing and relatively harmless it's just a better practice in general.